src/snowflake/cli/plugins/object/stage/manager.py

```diff
@@ -164,3 +186,90 @@ def create(self, stage_name: str, comment: Optional[str] = None) -> SnowflakeCur
     def iter_stage(self, stage_path: str):
         for file in self.list_files(stage_path).fetchall():
             yield file["name"]
+
+    def execute(
+        self,
+        stage_path: str,
+        on_error: OnErrorType,
+        variables: Optional[List[str]] = None,
+    ):
+        all_files_list = self._get_files_list_from_stage(stage_path)
+        # filter files from stage if match stage_path pattern
+        filtered_file_list = self._filter_files_list(stage_path, all_files_list)
+
+        if not filtered_file_list:
+            raise ClickException(f"No files matched pattern '{stage_path}'")
+
+        # sort filtered files in alphabetical order with directories at the end
+        sorted_file_list = sorted(
+            filtered_file_list, key=lambda f: (path.dirname(f), path.basename(f))
+        )
+
+        sql_variables = self._parse_execute_variables(variables)
+        results = []
+        for file in sorted_file_list:
+            results.append(self._call_execute_immediate(file, sql_variables, on_error))
+
+        return results
+
+    def _get_files_list_from_stage(self, stage_path: str) -> List[str]:
+        stage_name = self.get_stage_name_from_path(stage_path)
+        files_list_result = self.list_files(stage_name).fetchall()
```

Comment:
How big this last can be? Do we want to fetchall or do fetch one by one? Something to think about for future

Reply:
`ls @stage` returns files in alphabetical order ignoring directories, so we need to have whole filtered list to execute files in right order.